packages/app-mobile/components/screens/Note.tsx

```diff
@@ -1033,14 +982,20 @@ class NoteScreenComponent extends BaseScreenComponent<Props, State> implements B
 	}
 
 	public async onAlarmDialogAccept(date: Date) {
-		const response = await checkPermissions(PermissionsAndroid.PERMISSIONS.POST_NOTIFICATIONS);
+		if (Platform.OS === 'android') {
+			const response = await checkPermissions(PermissionsAndroid.PERMISSIONS.POST_NOTIFICATIONS);
+
+			// The POST_NOTIFICATIONS permission isn't supported on Android API < 33.
+			// (If unsupported, returns NEVER_ASK_AGAIN).
+			// On earlier releases, notifications should work without this permission.
+			if (response === PermissionsAndroid.RESULTS.DENIED) {
+				logger.warn('POST_NOTIFICATIONS permission was not granted');
+				return;
+			}
+		}
 
-		// The POST_NOTIFICATIONS permission isn't supported on Android API < 33.
-		// (If unsupported, returns NEVER_ASK_AGAIN).
-		// On earlier releases, notifications should work without this permission.
-		if (response === PermissionsAndroid.RESULTS.DENIED) {
-			logger.warn('POST_NOTIFICATIONS permission was not granted');
-			return;
+		if (Platform.OS === 'web') {
+			alert('Warning: The due-date has been saved, but showing notifications is not supported by Joplin Web.');
 		}
```

Comment:
We might be able to use the [Web Notification API](https://developer.mozilla.org/en-US/docs/Web/API/Notifications_API/Using_the_Notifications_API) for this. However, the app might need to be open for this to work (alternatively, we could use the [background sync API](https://developer.mozilla.org/en-US/docs/Web/API/Background_Synchronization_API), but this might not be accurate enough for alarm scheduling).